### ISSUE TITLE
Cannot install the python package on WSL or on Windows

### ISSUE THREAD

**Issue by giuliofoletto** [Tomographer/tomographer#8]
I am trying to install the tomographer python package via pip, but I cannot find an easy way to do so on Windows. Docs state indeed that the python package has not been tested on Windows, moreover, the conda packages `gcc` and `libgcc` are required, but these do not exist for Windows. 
Therefore I switched to my WSL installation (ubuntu 16.04) and installed the necessary packages using miniconda. As stated in the main readme of this repo, I used
`> conda install numpy gcc libgcc`
`> conda install -c conda-forge pybind11`
(notice that I also had to install gcc using apt-get). These installations seem to work fine. However the installation of tomographer itself fails, giving me all sorts of errors depending on the flags and options. I am attaching the log I get using
`CC=~/miniconda/envs/tomoenv/bin/gcc CXX=~/miniconda/envs/tomoenv/bin/g++ CXX_FLAGS="-stdlib=libc++" OPTIMIZATION_CXX_FLAGS="" pip install tomographer --verbose > log.txt`
Notice that `~/miniconda/envs/tomoenv` is the folder of the conda environment I created for tomographer and where I installed the above dependencies. 

Can you help me use the tomographer package either on WSL or Windows?

Thank you and best regards

[log.txt](https://github.com/Tomographer/tomographer/files/2540488/log.txt)

**Comment by giuliofoletto** [comment]
Ok I managed to install everything in WSL. The key is using the right C++ compiler. This is what I have done in WSL (ubuntu 16.04) and in a fresh conda environment
```bash
conda install numpy
conda install -c=conda-forge pybind11
conda install gcc_linux-64
conda install gxx_linux-64
pip install tomographer
```
Notice that I installed `gcc_linux-64` and `gxx_linux-64` and NOT `gcc` or `libgcc`. 
In order to run the examples I also had to install `scipy` and `matplotlib`
```bash
conda install scipy matplotlib
```
I suppose this is the right procedure on any Linux distro using conda, not just WSL. It looks like older packages such as `gcc` might be problematic. See also [the anaconda blog](https://www.anaconda.com/blog/developer-blog/utilizing-the-new-compilers-in-anaconda-distribution-5/) for more info.
For users on WSL who also need to show plots using `matplotlib` [this stackoverflow thread](https://stackoverflow.com/questions/43397162/show-matplotlib-plots-in-ubuntu-windows-subsystem-for-linux) might help.

**Comment by phfaist** [comment]
Ok, great that it worked out! I don't have a reliable windows workstation to do any testing, which is why I didn't optimize the windows installation experience. I've had several issues with compilers during testing on linux/mac, too, especially since there are a lot of different python environments and it's hard to support all combinations with different compilers etc. (linux vs mac, conda vs pip, gcc vs llvm, etc.). During testing it seemed that using conda's gcc was more reliable (i.e., ran into less errors) than external compilers, so that's the reason the readme says to use conda's gcc. But that's by no means mandatory.

**Comment by giuliofoletto** [comment]
If I find the time, I'll test the installation of the tomographer package on another ubuntu machine next week. As for Windows... I am not expert enough to find a solution. I suggest Windows 10 users install WSL, miniconda in it, and follow the procedure described above. (The actual code can still be written using Windows editors).

Thank you for this package and the paper.

**Comment by phfaist** [comment]
Great, thanks for your feedback!

**Comment by giuliofoletto** [comment]
I installed tomographer on another WSL machine running ubuntu 16.04 and also a full ubuntu machine running 18.04. I could not install it using the conda packages `gcc` and `libgcc`. I had to install `gcc_linux-64` and `gxx_linux-64`. I am pretty sure it is just a matter of environment variables, but I could not find the right values to make the installation work without those two packages. 

Notice that for some reason, installing only `gcc_linux-64` did not work, although it worked in the first WSL machine I tried 3 days ago. I edited the above post to reflect this. gcc works in mysterious ways.
